src/SafeProvider.ts

```diff
@@ -32,8 +34,9 @@ export type SafeProviderProps = {
 const queryClient = new QueryClient()
 
 export function SafeProvider(params: React.PropsWithChildren<SafeProviderProps>) {
-  const [initialized, setInitialized] = useState(false)
   const [config, setConfig] = useState(params.config)
+  const [isInitialized, setIsInitialized] = useState(false)
```

Comment:
Will the difference between isInitialized and isConnecting be clear for the final dev using the hooks?

Reply:
I think at least the difference between `isInitialized` and `isConnected` can be understood.
* **isInitialized**: True once both public and signer clients are initially set.
* **isConnected**: Only true, if a signer is set und the signer client is initialized.

But regarding the third flag isConnect**ing** I'm not sure as well, especially since its name is very similar to the isConnect**ed** flag and might be confusing. The idea is that *isConnecting* true, while a signer is in the process of connecting. But this state could also easily be defined by the consumer app when calling the `connect` function which returns a promise. So I think we could remove the `isConnecting` flag from the library, WDYT?